src/visualizer.js

```diff
@@ -186,3 +187,96 @@ if (config.visualizer.enabled) {
     },
   };
 }
+
+global.visualizer.myRoomDatasDraw = function(roomName) {
+  const fontSize = 0.65;
+  const room = Game.rooms[roomName];
+  const color = (coeff) => `rgb(${Math.floor(-(coeff - 1) * 255)},${Math.floor(coeff * 255)},0)`;
+  const stringsLabels = [{string: `Energie Average :`, pos: 3.8},
+    {string: `Stored Energy :`, pos: 3.2},
+    {string: `Queue length :`, pos: 3.8},
+    {string: `RCL ${room.controller.level} progress :`, pos: 3.8}
+  ];
+
+  let y = 0;
+  for (let label of stringsLabels) {
+    room.visual.text(label.string, label.pos, 0.75 + 2 * y * fontSize, {
+      color: 'rgb(255,255,255)',
+      font: fontSize,
+    });
+    y++;
+  }
+
+  const energy = (room.memory.energyStats && room.memory.energyStats.average) ||
+    room.energyAvailable;
+  room.visual.text(`${energy}`, 9, 1.5 * fontSize, {
+    color: color(energy / room.energyCapacityAvailable),
+    font: fontSize * 2,
+  });
+
+  room.visual.text(`${room.storage && room.storage.store[RESOURCE_ENERGY]}`, 9, 3.5 * fontSize, {
+    color: color(Math.min(room.storage && room.storage.store[RESOURCE_ENERGY], 500000) / 500000),
+    font: fontSize * 2,
+  });
+  room.visual.text(`${room.memory.queue.length}`, 9, 5.5 * fontSize, {
+    color: color((20 - Math.min(room.memory.queue.length, 20)) / 20),
+    font: fontSize * 2,
+  });
+  room.visual.text(`${Math.floor(100 * room.controller.progress / (room.controller.progressTotal || room.controller.progress))}%`, 9, 7.5 * fontSize, {
```

Comment:
this also showing ``NaN%`` in RCL 8

in RCL 8:
* ``room.controller.progressTotal === undefined``
* ``room.controller.progress === 0``

maybe ``Math.floor(100 * (room.controller.progressTotal ? (room.controller.progress / room.controller.progressTotal) : 1))``